avx512_gemm.h

```diff
@@ -324,9 +324,8 @@ struct AVX512_8bit {
       Integer pack4567 = Pack0123(sum4, sum5, sum6, sum7);
 
       auto total = PermuteSummer(pack0123, pack4567);
-      auto offset = A_rowidx * B_cols + B0_colidx;
-      auto result = inited_pipeline.run(total, offset);
-      writer(C, offset, result);
+      auto result = inited_pipeline.run(total, A_rowidx, B0_colidx);
+      writer(C, A_rowidx * B_cols + B0_colidx, result);
```

Comment:
Why isn't the writer just the last step of the pipeline?

Reply:
The writer is not a part of pipeline mostly because it's essential part of matrix multiplication whereas a postprocess is just an optional transformation of raw result. If we move it to a postprocess pipeline then every pipeline will have to contain writer as the last stage.

Reply:
Indeed but for all I know, I'll be writing tanh somewhere and sigmoid somewhere else after all the combobulation.  You also don't know if it will use one of the AVX512 compress-and-write instructions.

Reply:
Because of our new approach to handle result of multiplication i.e. callbacks, I think we can reject this pull request.